3rdparty/CMakeLists.txt

```diff
@@ -325,22 +325,46 @@ if(USE_SYSTEM_WOLFSSL)
 	add_library(wolfssl INTERFACE)
 	target_link_libraries(wolfssl INTERFACE PkgConfig::WolfSSL)
 else()
-	SET(BUILD_TESTS NO CACHE BOOL "Build test applications")
-	add_compile_definitions(HAVE_FFDHE_2048 TFM_TIMING_RESISTANT ECC_TIMING_RESISTANT
-					WC_RSA_BLINDING HAVE_AESGCM WOLFSSL_SHA512 WOLFSSL_SHA384 NO_DSA HAVE_ECC TFM_ECC256 ECC_SHAMIR
-					NO_RC4 NO_HC128 NO_RABBIT WOLFSSL_SHA224 WOLFSSL_SHA3 WOLFSSL_SHAKE256 HAVE_POLY1305
-					HAVE_ONE_TIME_AUTH HAVE_CHACHA HAVE_HASHDRBG HAVE_TLS_EXTENSIONS HAVE_SNI HAVE_SUPPORTED_CURVES
-					HAVE_EXTENDED_MASTER NO_RC4 HAVE_ENCRYPT_THEN_MAC NO_PSK NO_MD4
-					WC_NO_ASYNC_THREADING OPENSSL_EXTRA WOLFSSL_DES_ECB WC_NO_HARDEN)
-	set(ENABLED_HARDEN OFF CACHE BOOL "Disable Harden")
+	# TODO(cjj19970505@live.cn)
+	# OPENSSL_EXTRA, WOLFSSL_DES_ECB and HAVE_SNI are unconfigurable from CMake cache.
+	# but they do have it in a TODO list (wolfssl/CMakeList, 1021)
+	add_compile_definitions(OPENSSL_EXTRA WOLFSSL_DES_ECB HAVE_SNI)
+
+	set(WOLFSSL_TLS13 "no" CACHE INTERNAL "")
+	set(WOLFSSL_HARDEN "no" CACHE INTERNAL "")
```

Comment:
You can remove this line, I can't for the life of me remember why I disabled it originally, I think I was getting a compilation error.
I tested on linux with that line removed(and harden is on by default) and it seems to work fine with both rpcn & autoupdater.